src/main/scala/beam/agentsim/infrastructure/power/PowerController.scala

```diff
@@ -0,0 +1,85 @@
+package beam.agentsim.infrastructure.power
+
+import beam.cosim.helics.BeamFederate
+import beam.sim.BeamServices
+import beam.sim.config.BeamConfig
+import org.slf4j.{Logger, LoggerFactory}
+
+import scala.util.control.NonFatal
+import scala.util.{Failure, Try}
+
+class PowerController(beamServices: BeamServices, beamConfig: BeamConfig) {
+
+  private val logger: Logger = LoggerFactory.getLogger(classOf[PowerController])
+
+  private[power] lazy val beamFederateMaybe: Option[BeamFederate] = Try {
+    logger.debug("Init PowerController resources...")
+    BeamFederate.loadHelics
+    BeamFederate.getInstance(beamServices)
+  }.recoverWith {
+    case e =>
+      logger.error("Cannot init BeamFederate: {}", e.getMessage)
+      Failure(e)
+  }.toOption
+
+  lazy val isConnectedToGrid: Boolean = beamFederateMaybe.isDefined
+
+  /**
+    * Publishes required power to the grid
+    *
+    * @param requiredPower power (in joules) over planning horizon
+    */
+  def publishPowerOverPlanningHorizon(requiredPower: Double, currentTime: Int): Unit = {
+    logger.debug("Sending power over planning horizon {} to the grid at time {}...", requiredPower, currentTime)
+    beamFederateMaybe
+      .map(_.publishPowerOverPlanningHorizon(requiredPower))
+      .getOrElse {
```

Comment:
This should be a match on None, the get is misleading since it does nothing.

Reply:
Sounds reasonable, replaced on `option.fold()()`